x-pack/plugins/licensing/server/routes/feature_usage.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { IRouter, StartServicesAccessor } from 'src/core/server';
+import { LicensingPluginStart } from '../types';
+
+export function registerFeatureUsageRoute(
+  router: IRouter,
+  getStartServices: StartServicesAccessor<{}, LicensingPluginStart>
+) {
+  router.get(
+    { path: '/api/licensing/feature_usage', validate: false },
+    async (context, request, response) => {
+      const [, , { featureUsage }] = await getStartServices();
+      return response.ok({
+        body: [...featureUsage.getLastUsages().entries()].reduce(
+          (res, [featureName, lastUsage]) => {
+            return {
+              ...res,
+              [featureName]: new Date(lastUsage).toISOString(),
```

Comment:
This may have already been discussed but I wonder if there could be any problems with using absolute dates. If clocks are not configured correctly, it could make making sense of this data harder from the Cloud side. One option could be to return a relative date, ie. how many seconds ago it was used.

Reply:
Same comment as https://github.com/elastic/kibana/pull/63549#discussion_r410013750, not sure if we can change the expected format here.

> One option could be to return a relative date, ie. how many seconds ago it was used

This has different drawbacks btw, such as being vulnerable to latency, and forcing to be processed on the fly or enhanced to not loose the time reference of the consumer call.

Reply:
Another problem could be that a user couldn't know in what timezone the timestamp was created. Not sure if it's a critical problem.

Reply:
I assumed it was not an issue, as the API consumption is just going to pool calls to the API to count the actual number of usages in a given period of time. TBH I would have returned a unix timestamp instead of an ISO date representation if the elasticsearch API counterpart was not returning this format.

If we want to have correct timezones, I can just change the feature usage service to store dates instead of unix timestamps, but once again, this is assuming the actual user/client and the kibana server are on the same timezone.

My main difficulty on this feature is that it's very unclear who has the power of decision on that API.

Reply:
> Also I would be expecting cloud instances to have properly synchronized clocks between their machines/vms/cris so this is probably a non-issue?

I think this is a safe enough assumption for its usage. There's always the potential for clock drift, but even if we're off a bit it won't have catastrophic effects.

Using an ISO-8601 format for dates is 👍 , they can be easily translated to a unix timestamp. `Date::toISOString` doesn't include the timezone offset, which is completely fine in this situation as the timezone that the server is running in is irrelevant.